### DISCUSSION TITLE
Get Transaction Status

### DISCUSSION THREAD

**Discussion by Tranduy1dol** [aptos-labs/aptos-developer-discussions#276] [General]
Hi, If i have a transaction hash, and an account who does this transaction, how can i check that if this transaction is finalized or not?

**Comment by 0xjunha** [comment]
You can call the [Aptos fullnode API](https://aptos.dev/nodes/aptos-api-spec/#/), `/transactions/by_hash/{txn_hash}` `GET`.

From the response,
If the `type` field is `user_transaction` and the `success` field is `true`, the transaction is finalized and succeeded.
If the `type` field is `user_transaction` and the `success` field is `false`, the transaction is finalized but failed.
If the `type` field is `pending_transaction`, the transaction is not yet finalized.


Alternatively, you can use the aptos typescript sdk which calls the fullnode API internally:

- [getTransactionByHash](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/api/transaction.ts#L108) - to get transaction info
- [isPendingTransaction](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/api/transaction.ts#L130) - to check whether the transaction is pending

**Reply by Tranduy1dol** [reply]
I'm coding rust, is there any method to get transaction info like typescript sdk.
Thank's for your answer.

**Reply by 0xjunha** [reply]
If you're using Rust, you can use `aptos-rest-client` crate like: [example1](https://github.com/aptos-labs/aptos-core/blob/main/sdk/examples/transfer-coin.rs#L39) & [example2](https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos-rest-client/examples/account/main.rs#L28) and use [this function](https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos-rest-client/src/lib.rs#L880). this will call the same API internally.

**Reply by Tranduy1dol** [reply]
I decided to use the ```wait_for_transaction``` function. Maybe the original code of Aptos Core is complicated, so I want to see what kind of result this function returns. Does it return a status code, such as 1 for a successful transaction and 0 for a failed transaction?
So does this code work?
```
    async fn verify_inclusion(&self, external_id: &str) -> color_eyre::Result<DaVerificationStatus> {
        let client = self.client.clone();
        let txn_hash: PendingTransaction = external_id.parse().unwrap();
        let txn_response = client.wait_for_transaction(&txn_hash);
        match txn_response {
            None => Ok(DaVerificationStatus::Pending),
            Some(receipt) => match receipt.status_code {
                Some(status) if status == 1  => Ok(DaVerificationStatus::Verified),
                _ => Ok(DaVerificationStatus::Rejected),
            }
        }
    }
```
